include/eve/traits/overload/default_behaviors.hpp

```diff
@@ -93,18 +93,91 @@ namespace eve
     friend constexpr bool operator==(callable const&, callable const&)                { return true;  }
   };
 
+  template< template<typename> class Func
+  , typename OptionsValues
+  , typename... Options
+  >
+  struct relative_conditional_callable : callable<Func, OptionsValues, relative_conditional_option, Options...>
+  {
+    using base_t = callable<Func, OptionsValues, relative_conditional_option, Options...>;
+    using func_t =  Func<OptionsValues>;
+
+    struct ignore { template<typename T> operator T() { return T{}; } };
+
+    template<callable_options O, typename T, typename... Ts>
+    constexpr EVE_FORCEINLINE auto behavior(auto arch, O const& opts, T x0, Ts... xs) const
+    {
+      if constexpr (match_option<condition_key, O, ignore_none_>)
+      {
+        return base_t::behavior(arch, opts, x0, xs...);
+      }
+      else
+      {
+        // Grab the condition and drop it from the callable
+        auto[cond, rmv_cond] = opts.extract(condition_key);
+        using cond_t =  decltype(cond);
+        [[maybe_unused]] Func<decltype(rmv_cond)> const f{rmv_cond};
+
+        // Check that the mask and the value are of same kind if simd
+        constexpr bool compatible_mask = !(simd_value<decltype(cond.mask(as(x0)))> && scalar_value<decltype(f(x0, xs...))>);
```

Comment:
eh?

it's relative_conditional_expr. We know all of them. `cond.mask` works correctly. What are you trying to do here